panic: runtime error: invalid memory address or nil pointer dereference

### Checklist

- [X] This is a bug report, not a question. Ask questions on [discuss.ipfs.tech](https://discuss.ipfs.tech/c/help/13).
- [X] I have searched on the [issue tracker](https://github.com/ipfs/kubo/issues?q=is%3Aissue) for my bug.
- [X] I am running the latest [kubo version](https://dist.ipfs.tech/#kubo) or have an issue updating.

### Installation method

third-party binary

### Version

```Text
0.29.0
```


### Config

_No response_

### Description

```
ERROR cmd/ipfs        kubo/daemon.go:595      failed to access config: cannot access config, repo not open
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x1b0 pc=0x1bd9aca]

goroutine 9480 [running]:
github.com/ipfs/kubo/cmd/ipfs/kubo.daemonFunc.func7()
        github.com/ipfs/kubo@v0.29.0/cmd/ipfs/kubo/daemon.go:597 +0xaa
created by time.goFunc
        time/sleep.go:177 +0x2d
```
its bug in demon.go. you print error message but use returned value nil anyway

--- a/cmd/ipfs/kubo/daemon.go
+++ b/cmd/ipfs/kubo/daemon.go
@@ -597,6 +597,7 @@ take effect.
 			cfg, err := cctx.GetConfig()
 			if err != nil {
 				log.Errorf("failed to access config: %s", err)
+				return
 			}
 			if len(cfg.Bootstrap) == 0 && len(cfg.Peering.Peers) == 0 {
 				// Skip peer check if Bootstrap and Peering lists are empty
@@ -607,10 +608,12 @@ take effect.
 			ipfs, err := coreapi.NewCoreAPI(node)
 			if err != nil {
 				log.Errorf("failed to access CoreAPI: %v", err)
+				return
 			}
 			peers, err := ipfs.Swarm().Peers(cctx.Context())
 			if err != nil {
 				log.Errorf("failed to read swarm peers: %v", err)
+				return
 			}
 			if len(peers) == 0 {
 				log.Error("failed to bootstrap (no peers found): consider updating Bootstrap or Peering section of your config")

